examples/api-gateway/main.cpp

```diff
@@ -0,0 +1,54 @@
+#include <aws/lambda-runtime/runtime.h>
+#include <aws/core/utils/json/JsonSerializer.h>
+#include <aws/core/utils/memory/stl/SimpleStringStream.h>
+ 
+using namespace aws::lambda_runtime; 
+
+invocation_response my_handler(invocation_request const& request) 
+{
+	
+	using namespace Aws::Utils::Json;
+	
+	JsonValue json(request.payload);
+	if (!json.WasParseSuccessful()) {
+        	return invocation_response::failure("Failed to parse input JSON", "InvalidJSON");
+    	}
+
+	auto v = json.View();
+	Aws::SimpleStringStream ss; 
+	ss << "Good "; 
+
+	// see precondition remark in README 
+	if (v.ValueExists("body")) {
```

Comment:
Please add the validation and remove the remark.

The pattern is 
```c++
if (!v.ValueExists(key)) {
  // report error
}
auto key_view = v.GetObject(key);
if (!key_view.IsString()) {
 // report error
}
ss << key_view.GetString(); // safe call
```

Reply:
In keeping with the [NodeJS](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-create-api-as-simple-proxy-for-lambda.html) example, I opted to only return an error in the event that the initial parse fails. Additionally, I used ternary throughout for brevity, but if you think this decreases readability I would be happy to change.